### PR TITLE
autoload bazel-mode

### PR DESCRIPTION
**Pull request by condy0919**
#17 hasn't updated for a long time. So I opened a new pull request applying @phst advice in #17

### CODE REVIEW

File: bazel-mode.el [hunk 1]
```diff
@@ -107,6 +107,10 @@
   (add-hook 'before-save-hook #'bazel-mode--buildifier-before-save-hook
             nil :local))
 
+;;;###autoload
+(add-to-list 'auto-mode-alist
+             (cons (rx (or "BUILD" "WORKSPACE" ".bazel" ".bzl" ".BUILD") eos) #'bazel-mode))
```
Comment (phst) [inline bazel-mode.el]: You should probably start these with `/` so that files like `foobarBUILD` aren't matched.

Comment (condy0919) [reply bazel-mode.el]: The prefix "/" works like "^BUILD$" but I can't find document about the usage.
Could you provide some manual about this trick?

Comment (phst) [reply bazel-mode.el]: The keys in `auto-mode-alist` are matched against the entire expanded filename. Starting a match with `/` just matches the directory separator. The manual entry for `auto-mode-alist` has a few examples for this.

Comment (condy0919) [reply bazel-mode.el]: Oh, it makes sense. Thanks :)

**Pull request merged**